From aa9b3f89845c562a686cb4ae6824b1bb9789451a Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Fri, 25 Sep 2020 16:35:03 -0400
Subject: [PATCH 0771/1921] gpio: thunderx: fix irq allocation of MMC CD GPIO
 pin

The call to irq_chip_request_resources_parent() returns -ENOSYS
if the parent does not include any resources.  This is not an
error per-se - handle this condition.
This occurs when the MMC core 'mmc_start_host()' invokes
'mmc_gpiod_request_cd_irq()'.

This addresses the following issues:
  1. dmesg error
   genirq: Failed to request resources for \
    878000000000.pci:mrml-bridge0@1,0:mmc@1,4:mmc-slot@0 \
    cd (irq 1732) on irqchip GPIO
  2. the MMC host->caps property 'MMC_CAP_NEEDS_POLL' is set

Change-Id: I7307d715db4f7be949cff72be2bc4293566ba6ea
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/36739
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index bcf728b7cac6..c65d25068fbd 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -727,6 +727,9 @@ static int thunderx_gpio_irq_request_resources(struct irq_data *data)
 		return r;
 
 	r = irq_chip_request_resources_parent(data);
+	/* If parent does not HAVE resources, this isn't a failing error */
+	if (r == -ENOSYS)
+		r = 0;
 	if (r)
 		gpiochip_unlock_as_irq(&txgpio->chip, txline->line);
 
-- 
2.31.1

